create-post/create-post.js

```diff
@@ -0,0 +1,66 @@
+/* Imports */
+import '../auth/user.js';
+import { createPost, uploadImage } from '../fetch-utils.js';
+
+/* Get DOM Elements */
+const postForm = document.getElementById('post-form');
+const errorDisplay = document.getElementById('error');
+const addBtn = document.getElementById('add-post');
+const imageInput = document.getElementById('image-input');
+const preview = document.getElementById('preview');
+
+/* State */
+let error = null;
+
+/* Events */
+imageInput.addEventListener('change', () => {
+    const file = imageInput.files[0];
+    if (file) {
+        preview.src = URL.createObjectURL(file);
+    } else {
+        preview.src = '../assets/image-gallery.png';
+    }
+});
+
+postForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    addBtn.disabled = true;
```

Comment:
good on you for starting to use this kind of logic — it makes for a far smoother UX and when we get to react apps using loading state and changing the view based on it becomes very important